code-server/run.sh

```diff
@@ -10,6 +10,9 @@ CODE_SERVER="${INSTALL_PREFIX}/bin/code-server"
 EXTENSION_ARG=""
 if [ -n "${EXTENSIONS_DIR}" ]; then
   EXTENSION_ARG="--extensions-dir=${EXTENSIONS_DIR}"
+  EXTENSIONS_DIR="${EXTENSIONS_DIR}"
+else
+  EXTENSIONS_DIR="${INSTALL_PREFIX}/extensions"
```

Comment:
This may not be the default extensions directory, right?  It should be something like `$XDG_DATA_HOME/code-server/extensions` (or `$HOME/.local` if `XDG_DATA_HOME` is not set).  At least, for Linux.  I am not sure if these modules can run in macOS and Windows.

We could always set `--extensions-dir` so we know where the extensions are, although I suppose that would not be a backwards compatible change.

Maybe we could just always install extensions regardless of whether the directory exists, since if an extension already exists code-server will be fine with that.  That would have the advantage of installing new extensions that are added to the template, too.

Reply:
True. Let me adjust to handle defaults better.

Reply:
Always installing is slower for restarting a workspace. So I would like to get some speed up on restart. 

I will see how much complexity it will add.

Reply:
Maybe. Only cache extensions if there is an explicit extensions directory set. 

Otherwise always install (and maybe handle caches where existing extensions are there). 


I will need to document test cases as I go.